bee-tangle/src/metadata.rs

```diff
@@ -183,14 +179,77 @@ impl Packable for MessageMetadata {
     fn unpack<R: Read + ?Sized>(reader: &mut R) -> Result<Self, Self::Error> {
         Ok(Self {
             flags: Flags::unpack(reader)?,
-            milestone_index: MilestoneIndex::unpack(reader)?,
+            milestone_index: Option::<MilestoneIndex>::unpack(reader)?,
             arrival_timestamp: u64::unpack(reader)?,
             solidification_timestamp: u64::unpack(reader)?,
             confirmation_timestamp: u64::unpack(reader)?,
-            cone_index: Option::<MilestoneIndex>::unpack(reader)?,
-            otrsi: Option::<MilestoneIndex>::unpack(reader)?,
-            ytrsi: Option::<MilestoneIndex>::unpack(reader)?,
+            otrsi: Option::<IndexId>::unpack(reader)?,
+            ytrsi: Option::<IndexId>::unpack(reader)?,
             conflict: u8::unpack(reader)?,
         })
     }
 }
+
+#[derive(Clone, Copy, Debug)]
+pub struct IndexId(pub MilestoneIndex, pub MessageId);
```

Comment:
Can we make the fields private and have getters ? From an external point of view `.0` and `.1` are kinda confusing, you'd have to always go back to the type to know which one is which one. `.index()` and `.message_id()` would help a lot imo.

Reply:
agree, will change

Reply:
done